syntax = "proto3";

package ml;

service ml {
    rpc GetUsageData (UsageData_Request) returns (UsageData_Reply) {}
    rpc GetPredictionData (PredictionData_Request) returns (PredictionData_Reply) {}
}

message UsageData_Request {
    string householdID =1;
    int32 days = 2;
}

message UsageData {
    string timestamp = 1;
    float electric_usage = 2;
}

message UsageData_Reply {
    repeated UsageData items = 1;
}

message PredictionData_Request  {
    string householdID = 1;
}

message PredictionDataIndividual {
    string timestamp = 1;
    float electric_usage = 2;
}

message PredictionDataHouseholdType {
    string timestamp = 1;
    float electric_usage = 2;
}

message  PredictionData_Reply {
    repeated PredictionDataIndividual item = 1;
    repeated PredictionDataHouseholdType item2 = 2;
}
